cypress/e2e/room/room-header.spec.ts

```diff
@@ -0,0 +1,182 @@
+
+            cy.findByRole("button", { name: "Room options" }).should("be.visible");
+            cy.findByRole("button", { name: "Voice call" }).should("be.visible");
+            cy.findByRole("button", { name: "Video call" }).should("be.visible");
+            cy.findByRole("button", { name: "Search" }).should("be.visible");
+            cy.findByRole("button", { name: "Threads" }).should("be.visible");
+            cy.findByRole("button", { name: "Notifications" }).should("be.visible");
+            cy.findByRole("button", { name: "Room info" }).should("be.visible");
+        });
+
+        cy.get(".mx_RoomHeader").percySnapshotElement("Room header", {
+            widths: [300, 600], // Magic numbers
+        });
+    });
+
+    it("should render the pin button for pinned messages card", () => {
+        cy.enableLabsFeature("feature_pinning");
+
+        cy.createRoom({ name: "Test Room" }).viewRoomByName("Test Room");
+
+        cy.getComposer().type("Test message{enter}");
+
+        cy.get(".mx_EventTile_last").realHover().findByRole("button", { name: "Options" }).click();
+
+        cy.findByRole("menuitem", { name: "Pin" }).should("be.visible").click();
+
+        cy.get(".mx_RoomHeader").within(() => {
+            cy.findByRole("button", { name: "Pinned messages" }).should("be.visible");
+        });
+    });
+
+    it("should render a very long room name without collapsing the buttons", () => {
+        const LONG_ROOM_NAME =
+            "Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore " +
+            "et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut " +
+            "aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum " +
+            "dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui " +
+            "officia deserunt mollit anim id est laborum.";
+
+        cy.createRoom({ name: LONG_ROOM_NAME }).viewRoomByName(LONG_ROOM_NAME);
+
+        cy.get(".mx_RoomHeader").within(() => {
+            // Wait until the room name is set
+            cy.get(".mx_RoomHeader_nametext").within(() => {
+                cy.findByText(LONG_ROOM_NAME).should("exist");
+            });
+
+            // Assert the size of buttons on RoomHeader (except mx_RoomHeader_name) are specified
+            // and the buttons are not compressed
+            cy.get(".mx_RoomHeader_button")
+                .should("have.length", 3)
+                .should("be.visible")
+                .should("have.css", "height", "32px")
+                .should("have.css", "width", "32px");
+            cy.get(".mx_RightPanel_headerButton") // TODO: use the same class name
+                .should("have.length", 3)
+                .should("be.visible")
+                .should("have.css", "height", "32px")
+                .should("have.css", "width", "32px");
```

Comment:
How come we have a test for 7 buttons earlier on, but we only check 6 here?

Reply:
I'm going to add a comment about it. It is indeed confusing.

Reply:
Addressed with https://github.com/matrix-org/matrix-react-sdk/pull/10713/commits/de2dd1c4e1fc09ee5cdb92baa38fbd5172417ea2